pkg/linter/linter.go

```diff
@@ -178,6 +191,20 @@ func optLinter(_ LinterContext, path string, _ fs.DirEntry) error {
 
 	return nil
 }
+func objectLinter(_ LinterContext, path string, _ fs.DirEntry) error {
+	if filepath.Ext(path) == ".o" {
+		return fmt.Errorf("Package contains intermediate object file '%s', this is usually wrong", path)
```

Comment:
I would add a note saying how to resolve the issue. I would note that `.o` files should usually be removed.